### PR TITLE
[connecitivity_macos] Fix warnings to publish

### PR DESCRIPTION
**Pull request by franciscojma86**
## Description

Fix according to `flutter pub publish --dry-run` feedback

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [ ] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

### CODE REVIEW

**Comment by collinjackson** [comment]
I'm curious why `./script/check_publish.sh` didn't detect this. Can we improve it?

File: packages/connectivity/connectivity_macos/pubspec.yaml [hunk 1]
```diff
@@ -12,7 +11,7 @@ flutter:
 
 environment:
   sdk: ">=2.1.0 <3.0.0"
-  flutter: ">=1.9.1+hotfix.6 <2.0.0"
+  flutter: "^1.10.0"
```
Comment (collinjackson) [inline packages/connectivity/connectivity_macos/pubspec.yaml]: Did you mean to remove the <2.0.0 here?

Comment (franciscojma86) [reply packages/connectivity/connectivity_macos/pubspec.yaml]: I don't think I can use ^ and <2.0.0 together

**Comment by franciscojma86** [comment]
./script/check_publish.sh doesn't catch it while on stable.

**Pull request merged**